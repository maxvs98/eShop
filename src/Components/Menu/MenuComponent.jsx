import React from 'react';
import {
  Popup, List, Button, Image, Icon,
} from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from './img/logo.png';
import AuthButton from './AuthButton/AuthButtonContainer';

const CartComponent = ({
  title, id, picture, removeFromCart,
}) => (
  <List selection divided verticalAlign="middle">
    <List.Item>
      <List.Content floated="right">
        {/* eslint-disable */}
        <Button onClick={removeFromCart.bind(this,id)} color="red">Удалить</Button>
        {/* eslint-enable */}
      </List.Content>
      <Image avatar src={picture} />
      <List.Content>{title}</List.Content>
    </List.Item>
  </List>
);

CartComponent.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  picture: PropTypes.string.isRequired,
  removeFromCart: PropTypes.func.isRequired,
};

const MenuComponent = ({
  totalPrice, count, items, log,
}) => (
  <div>
    <div className="container">
      <div className="row">
        <div className="col-md-1 text-center">
          <a href="/">
            <div className="menuIcon" />
          </a>
        </div>
        <div className="col-md-3 text-center">
          <Link to="/">
            <img alt="logo" src={logo} />
          </Link>
        </div>
        <div className="col-md-5">
          <div className="menu clearfix">
            <Link to="/">
              <div className="menu__item">
                Home
              </div>
            </Link>
            <Link to="/shop">
              <div className="menu__item">
                Show goods
              </div>
            </Link>
            <Link to="/users">
              <div className="menu__item">
                Users
              </div>
            </Link>
          </div>
        </div>
        <div className="col-md-2">
          <div className="cart clearfix">
            <Popup
              trigger={
                (
                  <div>
                    <div className="cart__icon" />
                    <div className="cart__text">CART</div>
                    <div className="cart__counter">{count}</div>
                  </div>
                )
                }
              header={
                (
                  <p>
                    {totalPrice}
                    <Icon name="dollar sign" />
                  </p>
              )
              }
              content={
                /* eslint-disable */
                  items.map(product => <CartComponent {...product}/>)
                /* eslint-enable */
                }
              on="click"
              hideOnScroll
              position="bottom center"
              offset="0, 40px"
            />
          </div>
        </div>
        <div className="col-md-1 text-center">
          <div className="role__text">{log}</div>
          <AuthButton />
        </div>
      </div>
    </div>
  </div>
);

MenuComponent.propTypes = {
  totalPrice: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  items: PropTypes.shape.isRequired,
  log: PropTypes.string.isRequired,
};

export default MenuComponent;
