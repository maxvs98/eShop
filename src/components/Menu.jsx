import React from 'react';
import { Menu, Popup, List, Button, Image } from 'semantic-ui-react';

const CartComponent = ({title, id, picture, removeFromCart}) => (
  <List selection divided verticalAlign='middle'>
    <List.Item>
      <List.Content floated='right'>
        <Button onClick={removeFromCart.bind(this,id)} color="red">Удалить</Button>
      </List.Content>
      <Image avatar src={picture} />
      <List.Content>{title}</List.Content>
    </List.Item>
  </List>
);

const MenuComponent = ({ totalPrice, count, items }) => (
  <Menu>
    <Menu.Item
      name='browse'
    >
      Магазин одежды
    </Menu.Item>

    <Menu.Menu position="right">
      <Menu.Item
        name='singup'
      >
        Итого: &nbsp; <b>{totalPrice}</b>
      </Menu.Item>

      <Popup
        trigger = {
          <Menu.Item name='help'>
            Корзина (<b>{count}</b>)
          </Menu.Item>
        }
        content={items.map(product => <CartComponent {...product}/>)}
        on="click"
        hideOnScroll
      />
    </Menu.Menu>
  </Menu>
);

export default MenuComponent;
