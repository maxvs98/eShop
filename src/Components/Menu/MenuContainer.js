import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import uniqBy from 'lodash/uniqBy';
import * as cartActions from '../../storage/actions/cart';
import Menu from './MenuComponent';

const mapStateToProps = ({ cart, authorization }) => ({
  totalPrice:
    parseFloat((cart.items.reduce((total, product) => total + product.price, 0)).toFixed(3)),
  count: cart.items.length,
  items: uniqBy(cart.items, (o) => o.id),
  log: authorization.login,
});

const mapDispatchToProps = (dispatch) => ({
  ...bindActionCreators(cartActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
