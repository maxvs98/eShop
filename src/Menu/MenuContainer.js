import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as cartActions from '../actions/cart';
import Menu from './MenuComponent';
import uniqBy from 'lodash/uniqBy';

const mapStateToProps = ({ cart, authorization }) => ({
  totalPrice: parseFloat((cart.items.reduce((total, product) => total + product.price, 0)).toFixed(3)),
  count: cart.items.length,
  items: uniqBy(cart.items, o => o.id),
  role: authorization.role
});

const mapDispatchToProps = dispatch => ({
   ...bindActionCreators(cartActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
