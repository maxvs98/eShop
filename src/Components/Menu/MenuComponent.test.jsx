import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Menu from './MenuContainer';

import createStore from '../../storage/store';

const store = createStore();

test('First snapshot test', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Menu />
    </Provider>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
