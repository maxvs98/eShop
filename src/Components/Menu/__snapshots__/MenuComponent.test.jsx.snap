// Jest Snapshot v1, https://goo.gl/fbAQLP
/* eslint-disable */
exports[`First snapshot test 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-1 text-center"
      >
        <a
          href="/"
        >
          <div
            className="menuIcon"
          />
        </a>
      </div>
      <div
        className="col-md-3 text-center"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            src="logo.png"
          />
        </a>
      </div>
      <div
        className="col-md-5"
      >
        <div
          className="menu clearfix"
        >


          <div />

        </div>
      </div>
      <div
        className="col-md-2"
      >
        <div
          className="cart clearfix"
        >
          <div
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="cart__icon"
            />
            <div
              className="cart__text"
            >
              CART
            </div>
            <div
              className="cart__counter"
            >
              0
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-1 text-center"
      >
        <div
          className="role__text"
        />
        <p>
          <a
            href="/login"
            onClick={[Function]}
          >
            <button
              className="auth-button"
              type="button"
            >
              Sign in
            </button>
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
